### PR TITLE
Feat: wallet connect details

### PR DESCRIPTION
**Pull request by Asmadek**
Need to check 
* src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx
* src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts

image

### CODE REVIEW

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  return (
+    <BaseModal
+      closeButton
+      contentClass=""
+      title={t('walletDetails.simpleTitle')}
+      isOpen={isModalOpen}
+      onClose={closeWowModal}
+    >
+      <div className="flex flex-col w-full">
+        <div className="flex items-center justify-between gap-x-2 py-5 px-5 border-b border-divider">
```
Comment (egor0798) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:66]: `py-5 px-5` => `p-5`

**Review by egor0798 — APPROVED** [review]

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    setTimeout(onClose, DEFAULT_TRANSITION);
+  };
+
+  const accountsList = useMemo(() => {
+    const chains = chainsService.getChainsData();
+    const sortedChains = chainsService.sortChains(chains);
+
+    const accountsList = sortedChains.reduce<AccountItem[]>((acc, c) => {
+      const account = accounts.find((a) => c.chainId === (a as WalletConnectAccount).chainId);
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:42]: Not that important right now, but if have time and come up with a solution that is not N*N, that would be awesome 🤩
Maybe create a map of `accounts/chainIds`

**Review by tuul-wq — CHANGES_REQUESTED** [review]
Suggestions

File: src/renderer/entities/walletConnect/model/wallet-connect-model.ts [hunk 3]
```diff
@@ -46,6 +46,14 @@ const $uri = createStore<string>('').reset(reset);
 const $accounts = createStore<string[]>([]).reset(reset);
 const $pairings = createStore<PairingTypes.Struct[]>([]).reset(reset);
 
+const _extendSessions = createEffect((client: Client) => {
+  const sessions = client.session.getAll();
+
+  sessions.forEach((s) => {
+    client.extend({ topic: s.topic }).catch((e) => console.warn(e));
```
[the remark this answers is not in the document]
Comment (Asmadek) [reply src/renderer/entities/walletConnect/model/wallet-connect-model.ts:53]: it's just internal effect, we don't need to wait while it fulfilled

File: src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+const $accounts = combine(walletSelectModel.$walletForDetails, walletModel.$accounts, (...args): Account[] => {
+  const [walletForDetails, accounts] = args;
+
+  if (!walletForDetails) return [];
+
+  return accounts.filter((account) => account.walletId === walletForDetails.id);
+});
+
+const $connected = combine($accounts, walletConnectModel.$client, (accounts, client): boolean => {
```
Comment (tuul-wq) [inline src/renderer/widgets/WalletDetails/model/wallet-provider-model.ts:16]:
```javascript
  const account = accounts[0];
  if (!client || !account || !accountUtils.isWalletConnect(account)) return false;
  
  // code
```

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+  const connected = useUnit(walletProviderModel.$connected);
+
+  const closeWowModal = () => {
+    toggleIsModalOpen();
+
+    setTimeout(onClose, DEFAULT_TRANSITION);
+  };
+
+  const accountsList = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (tuul-wq) [reply src/renderer/widgets/WalletDetails/ui/WalletConnectDetails.tsx:39]: Business logic is also about [data transformation](https://en.wikipedia.org/wiki/Business_logic#:~:text=Business%20logic%20is%20the%20portion%20of%20an%20enterprise%20system%20which%20determines%20how%20data%20is%20transformed%20or%20calculated%2C%20and%20how%20it%20is%20routed%20to%20people%20or%20software%20(workflow))

A good illustration of app's layers
image

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**